### PR TITLE
add reportActiveConfigurationProfile command

### PR DESCRIPTION
**Pull request by feerrenrut**
### Link to issue number:
Initially developed due to https://github.com/nvaccess/nvda/pull/9239#issuecomment-466984136

### Summary of the issue:
Even though the issue of announcing the current profile was resolved in #9239 reporting the current profile name may be useful more generally.

### Description of how this pull request fixes the issue:
A new command is introduced that allows the active configuration profile to be reported. 
Reports "normal configuration profile active" or "{profileName} configuration profile active"
No default gesture is assigned.

### Testing performed:
Checked that the correct name is reported when a configuration profile is active, and when the normal configuration is active.

### Change log entry:

New Features
` - Added a command to report the active configuration profile (#9325).`

### CODE REVIEW

**Comment by DrSooom** [comment]
> `def script_reportActiveConfigurationProfile(self, gesture):`

"(self, gesture):" » "(self,gesture):" (See a few lines below)

What global input gesture is used for this command? CTRL+Shift+P would be possible. And I guess that the message output will be via speech and braille together, right?

**Comment by leonardder** [comment]
You could consider assigning shift+NVDA+p as default.

**Comment by feerrenrut** [comment]
> "(self, gesture):" » "(self,gesture):" (See a few lines below)

Visually, code without spaces is harder to read. So while much of the code is formatted this way, though certainly not all, I hope we are able to start to transition.

> And I guess that the message output will be via speech and braille together, right?

Correct, although I have not tested with a braille display.

> You could consider assigning shift+NVDA+p as default.

Given the revised approach to #9325 (not changing the title text), this command might not be so urgent. If there is demand for it, we could still consider including it. However I think we should be cautious about assigning a default gesture.

**Comment by DrSooom** [comment]
@feerrenrut: Not only visually, also on a braille display. As I'm not familiar with Python, I didn't know that both methods are valid and equal. I thought it was a typo. Thanks for the explanation.

**Comment by JulienCochuyt** [comment]
> You could consider assigning shift+NVDA+p as default.

As @DrSooom proposed in https://github.com/nvaccess/nvda/pull/9325#issuecomment-467269673 I think NVDA+control+shift+p is more relevant as it would:
- be more coherent with NVDA+control+p for profile configuration dialog
- leave NVDA+shift+p for #9707, which in turn is coherent with NVDA+p for cycle through speech symbols level

Btw, I am not at all in favor of unassigned default gestures for commands that can be useful for beginners to intermediate users.

**Comment by feerrenrut** [comment]
The issue this PR attempted to address was resolved in the PR. However I think it could still be useful functionality, so I'll mark it as ready for review. Since sensible gestures are short supply I'd prefer to leave this unbound initially. I don't think that working with profiles is a beginner feature.

File: source/globalCommands.py [hunk 1]
```diff
@@ -1710,6 +1710,22 @@ def script_activateInputGesturesDialog(self, gesture):
...
+			# Translators: Message announced when the command to report the current configuration profile and
+			# the default configuration profile is active.
+			activeProfileMessage = _("normal configuration profile active")
+		else:
+			# Translators: Message announced when the command to report the current configuration profile
+			# is active. The placeholder '{profilename}' is replaced with the name of the current active profile.
+			activeProfileMessage = _("{profileName} configuration profile active").format(profileName=activeProfileName)
+		ui.message(activeProfileMessage)
+	# Translators: Input help mode message for the report current configuration profile command.
+	script_reportActiveConfigurationProfile.__doc__ = _("Reports the name of the current NVDA configuration profile")
```
Comment (lukaszgo1) [inline source/globalCommands.py:1726]: Have you considered using script decorator here?

**Review by LeonarddeR — APPROVED** [review]

**Comment by leonardder** [comment]
@lukaszgo1 has a good point.

File: source/globalCommands.py (hunk 1 above)
Comment (feerrenrut) [reply source/globalCommands.py]: I have pushed a new commit to use the decorator now.

**Comment by feerrenrut** [comment]
I have addressed the changes. The core concept was already reviewed. I'm going to merge this PR.

**Comment by Adriani90** [comment]
is this documented accordingly in the user guide?

**Comment by feerrenrut** [comment]
@Adriani90 Are all commands documented in the user guide? It's not a particularly interesting command in need of explanation, and is listed in the gestures dialog.

**Comment by Adriani90** [comment]
I guess not.
I just thought about a short sentence in the profiles section, saying that it is possible to assign a gesture for reporting current profile. In my view users should be aware of this possibility. Such a sentence makes sense for all unassigned commands in coresponding sections of the user guide.

Von meinem iPhone gesendet

**Comment by feerrenrut** [comment]
Then we have two places to maintain the list of commands, and two places to keep translations consistent. I don't think adding something to the user guide is a good way to make a new feature discoverable, it's main purpose is to explain confusing or complicated options. The changes file was updated, it's purpose is discoverability. Otherwise, for a listing of all commands users should consult the inputGestures dialog. To specifically make unassigned commands more discoverable, we should enhance the input gestures dialog with filters.

**Comment by Adriani90** [comment]
@feerrenrut I agree. If there is a filter implemented, then it would help to have a cathegory which displays only the unassigned gestures for example.

**Pull request merged**